### DISCUSSION TITLE
How to best use a .Net backend with Nx

### DISCUSSION THREAD

**Discussion by jvskriubakken** [nrwl/nx#6890] [General]
Hi

We have been using create-react-app for building our react frontend and .net at the backend, 
but are now wanting to use the very promising Nx.

What is the recommended approach when having a backend WEB-app written in .Net and C#? BTW we are aiming for a mono -repo containing more than one app.

1. Is it recommended to have the C# backend code reside within the same mono-repo? If so, where and how?

2. Or is it recommended to let the app be publishable and let the C# backend get the published npm?

Thanks for any guidance!

Jørund

**Comment by brandonroberts** [comment]
Hi @jvskriubakken,

1. Yes, you can host both frontend and backend code in the monorepo. React is already supported as a primary platform, and there is an Nx plugin for .Net maintained separately by Nx core team member @AgentEnder here:

https://github.com/nx-dotnet/nx-dotnet

**Comment by jvskriubakken** [comment]
Thanks @brandonroberts 
I am checking out that  nx-dotnet plugin!

**Comment by jvskriubakken** [comment]
Just want to add that we found it useful to use the nx-dotnet plugin.

Thanks @brandonroberts !
